Mathlib/CategoryTheory/Limits/Shapes/Types.lean

```diff
@@ -153,6 +153,14 @@ noncomputable def isInitialPunit : IsInitial (PEmpty : Type u) :=
   initialIsInitial.ofIso initialIso
 #align category_theory.limits.types.is_initial_punit CategoryTheory.Limits.Types.isInitialPunit
 
+/-- An object in `Type u` is initial if and only if it is empty. -/
+lemma initial_iff_empty (X : Type u) : Nonempty (IsInitial X) ↔ IsEmpty X := by
```

Comment:
Is the corresponding lemma for terminal objects already there?

Reply:
There is [CategoryTheory.Limits.Types.isTerminalEquivUnique](https://leanprover-community.github.io/mathlib4_docs/Mathlib/CategoryTheory/Limits/Shapes/Types.html#CategoryTheory.Limits.Types.isTerminalEquivUnique) which is stated in a slightly different form, because `Unique X` is not `Prop`-valued. One can deduce the analogous statement `Nonempty (IsTerminal X) ↔ Nonempty (Unique X)` by applying [Equiv.nonempty_congr](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Logic/Equiv/Defs.html#Equiv.nonempty_congr). I considered adding this as a corollary, but decided against it.